maps-compose-utils/src/main/java/com/google/maps/android/compose/kml/parser/KmlStyleParser.kt

```diff
@@ -0,0 +1,230 @@
+        }
+
+        /**
+         * Parses color value from KML data in AABBGGRR format and returns the hue value
+         *
+         * @param color Color in AABBGGRR format
+         * @return Float hue value from color
+         */
+        private fun parseKmlColor(color: String): Pair<Float, Float> {
+            val c = color.substringAfter('#')
+            val integerColor =
+                Color.parseColor("#${convertColorToAARRGGB(c)}")
+
+            val alpha = if (c.length > 6) {
+                val intValue = c.substring(0, 2).toInt(16)
+                intValue.toFloat() / 255f
+            } else 1f
+
+            return Pair(convertIntColorToHueValue(integerColor), alpha)
+        }
+
+
+        /**
+         * Converts an integer color value and returns the hue value
+         *
+         * @param color Color as integer
+         * @return Float hue value from color
+         */
+        internal fun convertIntColorToHueValue(color: Int): Float {
+            val hsvValues = FloatArray(HSV_VALUES)
+            Color.colorToHSV(color, hsvValues)
+            return hsvValues[HUE_VALUE]
+        }
+
+        /**
+         * Converts a color format of the form (AA)BBGGRR to (AA)RRGGBB.
+         * Any leading or trailing spaces in the provided string will be trimmed prior to conversion.
+         *
+         * @param color Color in (AA)BBGGRR format
+         * @return color in (AA)RRGGBB format
+         */
+        private fun convertColorToAARRGGB(color: String): String {
+            val trimmedColor = color.trim()
+            return if (trimmedColor.length > 6) {
+                trimmedColor.substring(0, 2) +
+                        trimmedColor.substring(6, 8) +
+                        trimmedColor.substring(4, 6) +
+                        trimmedColor.substring(2, 4)
+            } else {
+                trimmedColor.substring(4, 6) +
+                        trimmedColor.substring(2, 4) +
+                        trimmedColor.substring(0, 2)
+            }
+        }
+
+        /**
+         * Computes a random color given an integer. Algorithm to compute the random color can be found in
+         * https://developers.google.com/kml/documentation/kmlreference#colormode
+         *
+         * @param color Color represented as an integer
+         * @return Integer representing a random color
+         */
+        internal fun computeRandomColor(color: Int): Int {
+            val random = Random()
+            var red = Color.red(color)
+            var green = Color.green(color)
+            var blue = Color.blue(color)
+
+            if (red != 0) {
+                red = random.nextInt(red)
+            }
+            if (blue != 0) {
+                blue = random.nextInt(blue)
+            }
+            if (green != 0) {
+                green = random.nextInt(green)
+            }
+            if (red == 0 && blue == 0 && green == 0) {
+                red = random.nextInt(256)
```

Comment:
not sure maar moet t niet 255 zijn?

Reply:
de value die je geeft aan nextInt is 'until', dus genereert een nummer van 0 t/m 255